src/core/control/tools/StrokeHandler.cpp

```diff
@@ -363,20 +349,39 @@ void StrokeHandler::onButtonDoublePressEvent(const PositionInputData& pos) {
 
 void StrokeHandler::destroySurface() {
     if (surfMask || crMask) {
+        assert(!this->fullRedraw);
```

Comment:
What is this assert guarding against?

Reply:
Here for historic reasons :) Will get rid of it